### PR TITLE
[Merged by Bors] - feat(Data/Complex): add lemmas like `re_eq_abs`

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Complex/Basic.lean [hunk 1]
```diff
@@ -1092,6 +1092,28 @@ theorem abs_im_lt_abs {z : ℂ} : |z.im| < Complex.abs z ↔ z.re ≠ 0 := by
...
+@[simp]
+lemma abs_re_eq_abs {z : ℂ} : |z.re| = abs z ↔ z.im = 0 :=
+  not_iff_not.1 <| (abs_re_le_abs z).lt_iff_ne.symm.trans abs_re_lt_abs
+
+@[simp]
+lemma abs_im_eq_abs {z : ℂ} : |z.im| = abs z ↔ z.re = 0 :=
+  not_iff_not.1 <| (abs_im_le_abs z).lt_iff_ne.symm.trans abs_im_lt_abs
+
+@[simp]
+lemma re_eq_abs {z : ℂ} : z.re = abs z ↔ 0 ≤ z.re ∧ z.im = 0 :=
```
Comment (urkud) [inline Mathlib/Data/Complex/Basic.lean:1104]: RHS of this lemma can be formulated as `0 ≤ z` but that makes sense only if you open the `ComplexOrder` locale. Which form should I use? Similarly for the next two lemmas and `z ≤ 0`.

Comment (j-loreaux) [reply Mathlib/Data/Complex/Basic.lean:1104]: My preference would be to open the `ComplexOrder` locale, and that we should do this whenever it allows use to phrase things naturally. If we have enough examples of it throughout the library, maybe people will see the value of having this as a global instance.

**Comment by j-loreaux** [comment]
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**